pkg/source/server/server.go

```diff
@@ -0,0 +1,101 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"google.golang.org/grpc"
+
+	sourcepb "github.com/numaproj/numaflow-go/pkg/apis/proto/source/v1"
+	"github.com/numaproj/numaflow-go/pkg/info"
+	sourcesdk "github.com/numaproj/numaflow-go/pkg/source"
+)
+
+type server struct {
+	svc *sourcesdk.Service
+}
+
+// New creates a new server object.
+func New(
+	pendingHandler sourcesdk.PendingHandler,
+	readHandler sourcesdk.ReadHandler,
+	ackHandler sourcesdk.AckHandler,
```

Comment:
```suggestion
	udSource sourcesdk.Source,
```

pass a that implements the interface